packages/entrykit/src/utils/withFeeCache.ts

```diff
@@ -0,0 +1,25 @@
+import { Chain, createClient, http } from "viem";
+import { cachedFeesPerGas } from "../actions/cachedFeesPerGas";
+
+type WithFeeCacheOptions = {
+  refreshInterval?: number;
+};
+
+export function withFeeCache(chain: Chain, options: WithFeeCacheOptions = { refreshInterval: 10_000 }): Chain {
```

Comment:
```suggestion
export function withFeeCache<const chain extends Chain>(chain: chain, options: WithFeeCacheOptions = { refreshInterval: 10_000 }): chain {
```

just to keep strong types